src/components/Clock.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class Clock extends React.Component {
+  state = {
+    dateOut: new Date(),
+  };
+
+  componentDidMount() {
+    this.timerInterval = setInterval(() => {
+      this.setState({ dateOut: new Date() });
+      // eslint-disable-next-line
+       console.log(this.state.dateOut.toLocaleTimeString());
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.name !== this.props.name) {
+      // eslint-disable-next-line
+      console.log(`The Clock was renamed from ${prevProps.name} to ${this.props.name}`);
+    }
+  }
```

Comment:
You just only need to console.log the new name. So you can make it directly when clicking the button